pkgs/games/minetest/mods/build-minetest-mod.nix

```diff
@@ -0,0 +1,33 @@
+{stdenv}:
+
+rec {
+  buildMinetestMod =
+    attrs@{
+      name ? "${attrs.pname}",
+      namePrefix ? "",
+      src,
+      unpackPhase ? "",
+      configurePhase ? "",
+      buildPhase ? "",
+      preInstall ? "",
+      postInstall ? "",
+      path ? ".",
+      addonInfo ? null,
+      ...
+    }:
+    stdenv.mkDerivation (attrs // {
+        name = namePrefix + name;
```

Comment:
what about `pname = namePrefix + name`

and in the arguments:

```nix
, author ? null
, namePrefix ? if author == null then "${author}-" else ""
, version ? null
```

so that it's ready for contentdb stuff